src/main/java/org/opentripplanner/graph_builder/module/interlining/InterlineProcessor.java

```diff
@@ -125,66 +139,122 @@ private Multimap<TripPatternPair, TripPair> getInterlinedTrips(
-          TripPattern currPattern = patternForTripTimes.get(curr);
-          var fromStop = prevPattern.lastStop();
-          var toStop = currPattern.firstStop();
-          double teleportationDistance = SphericalDistanceLibrary.fastDistance(
-            fromStop.getLat(),
-            fromStop.getLon(),
-            toStop.getLat(),
-            toStop.getLon()
+          BitSet daysForToTripTimes = getDaysForService(
+            toTripTimes.getTrip().getServiceId(),
+            false
           );
-          if (teleportationDistance > maxInterlineDistance) {
-            issueStore.add(
-              new InterliningTeleport(
-                prev.getTrip(),
-                block.blockId(),
-                (int) teleportationDistance,
-                fromStop,
-                toStop
-              )
-            );
-            // Only skip this particular interline edge; there may be other valid ones in the block.
-          } else {
-            interlines.put(
-              new TripPatternPair(prevPattern, currPattern),
-              new TripPair(prev.getTrip(), curr.getTrip())
-            );
+          if (
+            uncoveredDays.intersects(daysForToTripTimes) &&
+            createInterline(fromTripTimes, toTripTimes, blockId, patternForTripTimes, interlines)
+          ) {
+            uncoveredDays.andNot(daysForToTripTimes);
+            if (uncoveredDays.isEmpty()) {
+              break;
+            }
           }
         }
-        prev = curr;
       }
     }
 
     return interlines;
   }
 
   /**
-   * This compound key object is used when grouping interlining trips together by (serviceId,
-   * blockId).
+   * Validates that trip times are continuous and that the transfer stop(s) are not too far away
+   * from each other. Then creates interline between the trips.
+   *
+   * @return true if interline has been created or if there is an issue preventing an interline
+   * creation for certain service dates.
    */
-  private record BlockIdAndServiceId(String blockId, FeedScopedId serviceId) {
-    static BlockIdAndServiceId ofTrip(Trip trip) {
-      return new BlockIdAndServiceId(trip.getGtfsBlockId(), trip.getServiceId());
+  private boolean createInterline(
+    TripTimes fromTripTimes,
+    TripTimes toTripTimes,
+    String blockId,
+    Map<TripTimes, TripPattern> patternForTripTimes,
+    Multimap<TripPatternPair, TripPair> interlines
+  ) {
+    if (
+      fromTripTimes.getDepartureTime(fromTripTimes.getNumStops() - 1) >
+      toTripTimes.getArrivalTime(0)
+    ) {
+      LOG.error(
+        "Trip times within block {} are not increasing on after trip {}.",
+        blockId,
+        fromTripTimes.getTrip().getId()
+      );
+      return true;
+    }
+    TripPattern fromPattern = patternForTripTimes.get(fromTripTimes);
+    TripPattern toPattern = patternForTripTimes.get(toTripTimes);
+    var fromStop = fromPattern.lastStop();
+    var toStop = toPattern.firstStop();
```

Comment:
```suggestion
    var fromPattern = patternForTripTimes.get(fromTripTimes);
    var toPattern = patternForTripTimes.get(toTripTimes);
    var fromStop = fromPattern.lastStop();
    var toStop = toPattern.firstStop();
```
Try to be consistent, for short scoped variables `var` is easier to read - unless the type is unclear.